File: src/NodeParser/VoidTypeNodeParser.ts
import ts from "typescript";
import type { Context } from "../NodeParser.js";
import type { SubNodeParser } from "../SubNodeParser.js";
import type { BaseType } from "../Type/BaseType.js";
import { VoidType } from "../Type/VoidType.js";

export class VoidTypeNodeParser implements SubNodeParser {
    public supportsNode(node: ts.KeywordTypeNode): boolean {
        return node.kind === ts.SyntaxKind.VoidKeyword;
    }
    public createType(node: ts.KeywordTypeNode, context: Context): BaseType {
        return new VoidType();
    }
}

File: internal/parser/void_type_node_parser.go
package parser

import (
	"github.com/microsoft/typescript-go/shim/ast"

	"github.com/vega/ts-json-schema-generator/internal/types"
)

// VoidTypeNodeParser handles the `void` keyword
// (src/NodeParser/VoidTypeNodeParser.ts).
type VoidTypeNodeParser struct{}

func NewVoidTypeNodeParser() *VoidTypeNodeParser {
	return &VoidTypeNodeParser{}
}

func (p *VoidTypeNodeParser) SupportsNode(node *ast.Node) bool {
	return node.Kind == ast.KindVoidKeyword
}

func (p *VoidTypeNodeParser) CreateType(_ *ast.Node, _ *Context, _ *types.ReferenceType) types.Type {
	return &types.VoidType{}
}
